net/src/net.cpp

```diff
@@ -89,6 +90,322 @@ int getAsync(lua_State* L)
     return lua_yield(L, 0);
 }
 
+static std::unordered_map<int, std::shared_ptr<uWS::App>> serverInstances;
+static std::unordered_map<int, std::shared_ptr<struct ServerLoopState>> serverStates;
+static int nextServerId = 1;
+
+struct ServerLoopState
+{
+    std::shared_ptr<uWS::App> app;
+    Runtime* runtime;
+    bool running = true;
+    std::function<void()> loopFunction;
+    std::shared_ptr<Ref> handlerRef;
+    std::string hostname;
+    int port;
+};
+
+static void parseQuery(const std::string& query, lua_State* L)
+{
+    lua_createtable(L, 0, 0);
+    size_t start = 0;
+    size_t end = query.find('&');
+    while (end != std::string::npos)
+    {
+        std::string pair = query.substr(start, end - start);
+        size_t eq = pair.find('=');
+        if (eq != std::string::npos)
+        {
+            std::string key = pair.substr(0, eq);
+            std::string value = pair.substr(eq + 1);
+            lua_pushstring(L, key.c_str());
+            lua_pushstring(L, value.c_str());
+            lua_settable(L, -3);
+        }
+        start = end + 1;
+        end = query.find('&', start);
+    }
+    std::string pair = query.substr(start);
+    size_t eq = pair.find('=');
+    if (eq != std::string::npos)
+    {
+        std::string key = pair.substr(0, eq);
+        std::string value = pair.substr(eq + 1);
+        lua_pushstring(L, key.c_str());
+        lua_pushstring(L, value.c_str());
+        lua_settable(L, -3);
+    }
+}
+
+static void parseHeaders(auto* req, lua_State* L)
+{
+    lua_createtable(L, 0, 0);
+    for (const auto& header : *req)
+    {
+        lua_pushlstring(L, header.first.data(), header.first.size());
+        lua_pushlstring(L, header.second.data(), header.second.size());
+        lua_settable(L, -3);
+    }
+}
+
+static void handleResponse(auto* res, lua_State* L, int responseIndex)
+{
+    // Check if the response is a string or a table
+    if (lua_isstring(L, responseIndex))
+    {
+        std::string body = lua_tostring(L, responseIndex);
+        res->writeStatus("200 OK");
+        res->writeHeader("Content-Type", "text/html");
+        res->end(body);
+    } 
+    else if (lua_istable(L, responseIndex))
```

Comment:
```suggestion
    // Check if the response is a string or a table
    if (lua_isstring(L, responseIndex))
    {
        std::string body = lua_tostring(L, responseIndex);
        res->writeStatus("200 OK");
        res->writeHeader("Content-Type", "text/html");
        res->end(body);
        return;
    }
 
    if (lua_istable(L, responseIndex))
```

generally prefer early returns.

i would also move the `500` server error up to an

`if (!lua_istable(L, responseIndex))` early exit as well, and then keep the main chunk of the implementation as the main flow of the function